File: spring-cloud-bus-demo/consumer-bus/src/main/java/com/livk/consumer/ConsumerBus.java
package com.livk.consumer;

import com.livk.bus.listener.EnableRemoteEventListener;
import com.livk.common.LivkSpring;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.cloud.client.discovery.EnableDiscoveryClient;

/**
 * <p>
 * ConsumerBus
 * </p>
 *
 * @author livk
 * @date 2021/11/1
 */
@Slf4j
@EnableDiscoveryClient
@EnableRemoteEventListener
@SpringBootApplication
public class ConsumerBus {

	public static void main(String[] args) {
		LivkSpring.run(ConsumerBus.class, args);
	}

}

File: spring-cloud-bus/consumer-bus/src/main/kotlin/com/livk/consumer/ConsumerBus.kt
package com.livk.consumer

import com.livk.bus.listener.EnableRemoteEventListener
import org.springframework.boot.ApplicationRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.cloud.client.discovery.EnableDiscoveryClient
import org.springframework.context.annotation.Bean
import org.springframework.core.env.Environment

/**
 *
 *
 * ConsumerBus
 *
 *
 * @author livk
 */
@EnableDiscoveryClient
@EnableRemoteEventListener
@SpringBootApplication
open class ConsumerBus {

    @Bean
    open fun env(env: Environment): ApplicationRunner {
        return ApplicationRunner {
            println(env)
        }
    }
}

fun main(args: Array<String>) {
    runApplication<ConsumerBus>(*args)
}
